### ISSUE TITLE
Add instrumentation for Kafka

### ISSUE THREAD

**Issue by rajan123456** [elastic/apm-agent-python#853]
**Is your feature request related to a problem? Please describe.**
The apm-agent-java is capable of tracing transactions through Kafka from v7.6, however, Kafka related messages cannot be tracked through the python agent.

**Describe the solution you'd like**
apm-agent-python should be able to support Kafka as part of the transaction details captured.

**Describe alternatives you've considered**
NA

**Additional context**
NA

**Comment by basepi** [comment]
Thanks for opening this! Kafka is definitely a high priority for us to support.

**Comment by Rohit-dot-KumarAM** [comment]
As mentioned here https://www.elastic.co/guide/en/apm/agent/python/current/how-the-agent-works.html#how-it-works-framework-integration.

Tried searching for any hooks/signals if available for **kafka** as it is there for **celery**, but couldn't find any 
If something similar for **kafka** exists, please share the info.

**Comment by basepi** [comment]
Yeah I don't think there are hooks, but we should be able to instrument the relevant pieces with our normal wrapping mechanism, and since modern kafka supports headers, we can now support distributed tracing as well. (Kafka support is already present in the Java agent, and we want to support it here as well.)

**Comment by Rohit-dot-KumarAM** [comment]
We did some effort to instrument at least **send()** from the **KafkaProducer** and **next()** from **KafkaConsumer** with  wrapping mechanism. But, we could do it only for django(When Kafka is being used as part of django Application).

**Comment by nhogan5** [comment]
Would love to see this to complete the link in my distributed tracing chain!
